README.md
## Announcements

Checkout out my [CLG EDITION: ES7 React/Redux/React-Native/JS snippets](https://marketplace.visualstudio.com/items?itemName=alexkev.clg-edition-es7-react-js-snippets). Its removes conflicting clg snippets with dsznajder's ES7 React/Redux/React-Native/JS snippets

## Features

Demo on [stackblitz](https://stackblitz.com/edit/clg?embed=1&file=index.ts&hideExplorer=1&hideNavigation=1&view=editor "a live demo of the clg snippets"). No install required! 👍 👍

![clg demo gif](images/clg.gif)

### Examples

|      Prefix | Method                                                                                                                                                                                                                                                                                                                            |
| ----------: | :-------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
|      `clg→` | `console.log("%cclg.js line:35 foo", "color: #007acc;", foo);`                                                                                                                                                                                                                                                                    |
|      `clgj→` | `console.log('%cAbout.tsx line:8 foo', 'color: #007acc;', JSON.stringify(foo, null, "\t" ))`                                                                                                                                                                                                                                                                    |
|     `clgg→` | `console.log("%cclg.js line:37 foo", "color: #26bfa5;", foo);`                                                                                                                                                                                                                                                                    |
|    `clgbb→` | `console.log("%cclg.js line:39 foo", "color: white; background-color: #007acc;", foo);`                                                                                                                                                                                                                                           |
|    `clggb→` | `console.log("%cclg.js line:41 foo", "color: white; background-color: #26bfa5;", foo);`                                                                                                                                                                                                                                           |
| `clghello→` | `console.log("%cHello clg.js", "background: green; color: white; display: block;");`                                                                                                                                                                                                                                              |
| `clgnhello→` | `console.log('\\x1b[42m%s\\x1b[0m', 'Hello ${TM_FILENAME} line:${TM_LINE_NUMBER}$1'`                                                                                                                                                                                                                                              |
|   `clgerr→` | `console.log("%cerror clg.js", "color: red; display: block; width: 100%;", error);`                                                                                                                                                                                                                                               |
|     `clgr→` | `console.log("%c Rainbowww!", "font-weight: bold; font-size: 50px;color: red; text-shadow: 3px 3px 0 rgb(217,31,38) , 6px 6px 0 rgb(226,91,14) , 9px 9px 0 rgb(245,221,8) , 12px 12px 0 rgb(5,148,68) , 15px 15px 0 rgb(2,135,206) , 18px 18px 0 rgb(4,77,145) , 21px 21px 0 rgb(42,21,113); margin-bottom: 12px; padding: 5%");` |
|   `clghuh→` | `console.log("%c ", "font-size: 1px; padding: 166.5px 250px; background-size: 500px 333px; background: no-repeat url(https://i.cloudup.com/Zqeq2GhGjt-3000x3000.jpeg);");`                                                                                                                                                        |
|     `clgg→` | `console.log("%c ", "font-size: 1px; padding: 240px 123.5px; background-size: 247px 480px; background: no-repeat url(https://i2.wp.com/i.giphy.com/media/11ZSwQNWba4YF2/giphy-downsized.gif?w=770&amp;ssl=1);");`                                                                                                                 |
| `clgclbrt→` | `console.log("%c ", "font-size: 1px; padding: 125px 125px; background-size: 250px 250px; background: no-repeat url(https://i2.wp.com/i.giphy.com/media/12BYUePgtn7sis/giphy-downsized.gif?w=770&amp;ssl=1);");`                                                                                                                   |
|     `clgc→` | `console.log("%c ", "font-size: 1px; padding: 215px 385px; background-size: 770px 430px; background: no-repeat url(https://i0.wp.com/i.giphy.com/media/ZVik7pBtu9dNS/giphy-downsized.gif?w=770&amp;ssl=1);");`                                                                                                                    |
### Complete Index

| Prefix | Description | 
| :---- | :------- | 
| clgbb | Simple Console Log Shorcut for JS Objects - Blue Background | 
| clgbug | Bugs, bugs, bugs and more bugs | 
| clgc | Coding GIF | 
| clgerr | Find errors with style | 
| clggb | Simple Console Log Shorcut for JS Objects - Green Background | 
| clgg | Simple Console Log Shorcut for JS Objects - Green Text | 
| clghello | Console logs ->  Hello ${TM_FILENAME} line:${TM_LINE_NUMBER} on Green Background | 
| clgclbrt | Celebrate your code works! | 
| clgnj | CLG: Node JSON String Output, Node ANSII Color Blue Background | 
| clgj | JSON String Output | 
| clgnbb | Node ANSII Color Blue Background | 
| clgnbfg | Node ANSII Color Blue Foreground | 
| clgngb | Node ANSII Color Green Background | 
| clgngfg | Node ANSII Color Green Foreground | 
| clgnrb | Node ANSII Color RED Background | 
| clgnrfg | Node ANSII Color RED Foreground | 
| clgnhello | Node Console logs ->  Hello ${TM_FILENAME} line:${TM_LINE_NUMBER} on Green Background | 
| clgr | You need a Rainbow in your code | 
| clghuh | Confused?! so put a this guy in your code. | 
| clg | Simple Console Log Shorcut for JS Objects - Blue Text | 
  
  
## Release Notes
Feel to free to fork my [repo](https://github.com/alexkev/clg) and if I like it I'll merge it in.

## Enjoy!

Support Open Source Code by buying me a drink ⚡🥤 😉.

[![paypal](https://www.paypalobjects.com/en_US/i/btn/btn_donateCC_LG.gif)](https://www.paypal.com/cgi-bin/webscr?cmd=_donations&business=axel720%40gmail.com&currency_code=USD&source=url)

### Credit:

Rainbow clg : [Mahesh Haldar](https://hackernoon.com/styling-logs-in-browser-console-2ec0807dc91a)

Background image in clg : [BrunoLM](https://stackoverflow.com/questions/26283936/stylized-console-logging)


CHANGELOG.md
# Change Log
-   1.3.2
    - A few more snippets and added a snippet index to docs (auto-genrated) 
-   1.2.0
    - ANSII Color Styling for Node.js
-   1.1.0
    - clg JSON output 
-   1.0.0
    - Anouncement: [CLG EDITION: ES7 React/Redux/React-Native/JS snippets](https://marketplace.visualstudio.com/items?itemName=alexkev.clg-edition-es7-react-js-snippets) 
    - Badges
-   0.1.2
    - Update file extension & language mode support
-   0.1.0
    -   clg - "console.log('%c$4${TM_FILENAME} ${5:line:${TM_LINE_NUMBER}} ' Object.keys({$1})[0]+ '$3', 'color: blue', \${1:Object});"
    -   replaced with: console.log('%c${TM_FILENAME} line:${TM_LINE_NUMBER} ${1:object}$2', 'color: #007acc;', \${1:object})
-   Initial release < 0.0.6


scripts/map-snippets-to-readme.js
const fs = require('fs').promises
const snippets = require("../snippets/snippets.json")

const errCallback = (err) => {
    if (err) throw err;
}

(async () => {
    await fs.writeFile("../README.md", '', errCallback)

    const start = await fs.readFile('./startReadMe.md');
    const end = await fs.readFile('./endReadMe.md');

    await fs.appendFile("../README.md", start, errCallback)

    await fs.appendFile("../README.md", `| Prefix | Description | \r\n| :---- | :------- | \r\n`, errCallback)

    const snippetSorted = await Object.fromEntries(
        Object.entries(snippets).sort()
    );

    for (const snip in snippetSorted) {
        if (Object.hasOwnProperty.call(snippetSorted, snip)) {
            const element = snippetSorted[snip];
            await fs.appendFile("../README.md", `| ${element.prefix} | ${element.description} | \r\n`, errCallback)
        }
    }

    await fs.appendFile("../README.md", end, errCallback)
})();

clg.js
/*
 **********************************
 *** FILE: clg.js
 *** Test and Document Snippets
 *** Press F12 then CLICK BOTH, at the bottom, to execute the ts file,
 *** then preview to return to editot
 **********************************
 */

/* These are the shortcuts! take them for a spin 
 * Press F12 to view console
clg
clgg
clgbb
clggb
clghello
clgerr
clgr
clghuh
clgg
clgclbrt
clgc 
*/

const car = {
    model: "ferrari",
    year: "2020",
};

const error = {
    results: false,
};

// clg - Simple Console Log Shorcut for JS Objects - Blue Text"
console.log("%cclg.js line:35 car", "color: #007acc;", car);
// clgg - Simple Console Log Shorcut for JS Objects - Green Text"
console.log("%cclg.js line:37 car", "color: #26bfa5;", car);
// clgbb - Simple Console Log Shorcut for JS Objects - Blue Background"
console.log("%cclg.js line:39 car", "color: white; background-color: #007acc;", car);
// clggb - Simple Console Log Shorcut for JS Objects - Green Background"
console.log("%cclg.js line:41 car", "color: white; background-color: #26bfa5;", car);

// clghello - Need to see it a function is being called? Use this: clghello
console.log("%cHello clg.js line:20 new code", "background: green; color: white; display: block;");
// clgerr - Find errors with style
console.log("%cerror clg.js line:22 here", "color: red; display: block; width: 100%;", error);
// clgr - You need a Rainbow in your code
console.log(
    "%c Alex was Here",
    "font-weight: bold; font-size: 50px;color: red; text-shadow: 3px 3px 0 rgb(217,31,38) , 6px 6px 0 rgb(226,91,14) , 9px 9px 0 rgb(245,221,8) , 12px 12px 0 rgb(5,148,68) , 15px 15px 0 rgb(2,135,206) , 18px 18px 0 rgb(4,77,145) , 21px 21px 0 rgb(42,21,113); margin-bottom: 12px; padding: 5%",
);
// clghuh - Confused?! so put a this guy in your code.
console.log(
    "%c ",
    "background: no-repeat url(https://i.cloudup.com/Zqeq2GhGjt-3000x3000.jpeg); font-size: 1px; padding: 166.5px 250px; background-size: 500px 333px;",
);
// clgg - Bugs, bugs, bugs and more bugs GIF
console.log(
    "%c ",
    "font-size: 1px; padding: 240px 123.5px; background-size: 247px 480px; background: no-repeat url(https://i2.wp.com/i.giphy.com/media/11ZSwQNWba4YF2/giphy-downsized.gif?w=770&amp;ssl=1);",
);
// clgclbrt - Celebrate! your code works! GIF
console.log(
    "%c ",
    "font-size: 1px; padding: 125px 125px; background-size: 250px 250px; background: no-repeat url(https://i2.wp.com/i.giphy.com/media/12BYUePgtn7sis/giphy-downsized.gif?w=770&amp;ssl=1);",
);
// clgc - Coding GIF
console.log(
    "%c ",
    "font-size: 1px; padding: 215px 385px; background-size: 770px 430px; background: no-repeat url(https://i0.wp.com/i.giphy.com/media/ZVik7pBtu9dNS/giphy-downsized.gif?w=770&amp;ssl=1);",
);

// ANSCII Colors
console.log('\x1b[44m%s\x1b[0m', 'clg.js line:73 car.model', car.model);
console.log('\x1b[34m%s\x1b[0m', 'clg.js line:74 car.model', car.model);

console.log(
    '%cclg.js line:77 car',
    'color: #007acc;',
    JSON.stringify(car, null, "\t")
);